src/app/parcels/parcelsTable/conditionalStyling.ts

```diff
@@ -24,19 +25,44 @@ const searchForPackingDateBreakPoints = (parcelsTableRows: ParcelsTableRow[]): n
     return getRowBoundaries(packingSlotValues);
 };
 
-export const searchForBreakPoints = (
-    sortField: string,
-    parcelsTableRows: ParcelsTableRow[]
-): number[][] => {
-    switch (sortField) {
-        case "packingSlot":
-            return [searchForPackingSlotBreakPoints(parcelsTableRows), []];
-        case "packingDate":
-            return [
-                searchForPackingDateBreakPoints(parcelsTableRows),
-                searchForPackingSlotBreakPoints(parcelsTableRows),
-            ];
+export type BreakPointConfig = {
+    name: string;
+    breakPoints: number[];
+    colour: string;
+    thickness: string;
+};
+
+export const searchForBreakPoints = <TableRow extends ParcelsTableRow>(
```

Comment:
can you explain the use of this type please? why TableRow extends ParcelsTableRow, why not just ParcelsTableRow?

Reply:
could just leave it as ParcelsTableRow tbf. i was thinking that if `searchForBreakPoints` will be used in the future for other tables it might be more easily generalisable? let me know if you want me to just leave it as `ParcelsTableRow` :)

Reply:
what this generic type currently does is it means that when we call this function, the type of the table row has to extend the type of ParcelsTableRow, this won't help us be more generalisable

also, searchForPackingDateBreakPoints takes in a type "ParcelsTableRow", and we refer to column headers from that type in the switch case.
so your function is only going to work for data of that type

I think there are 2 options: make your function work with ParcelsTableRow only, or make your function work with a generic TableRow type. your function currently only works with ParcelsTableRow, and i don't think it's worth the effort to get it to work with a generic type for now

I would just remove the generic and use ParcelsTableRow